layers/gpu/instrumentation/gpuav_instrumentation.cpp

```diff
@@ -432,25 +432,31 @@ void PostCallSetupShaderInstrumentationResources(Validator &gpuav, CommandBuffer
         std::shared_ptr<const vvl::PipelineLayout> last_bound_desc_set_pipe_layout_state =
             gpuav.Get<vvl::PipelineLayout>(last_bound.desc_set_pipeline_layout);
         if (last_bound_desc_set_pipe_layout_state) {
-            const uint32_t bindings_counts_from_last_pipeline =
-                GetBindingsCountFromLastBoundPipelineOrShader(gpuav, bind_point, cb_state, last_bound);
+            const uint32_t desc_set_bindings_counts_from_last_pipeline =
+                GetDescSetBindingsCountFromLastBoundPipelineOrShader(gpuav, bind_point, cb_state, last_bound);
 
-            const bool any_disturbed_bindings = bindings_counts_from_last_pipeline <
-                                                static_cast<uint32_t>(last_bound_desc_set_pipe_layout_state->set_layouts.size());
+            const bool any_disturbed_desc_sets_bindings =
+                desc_set_bindings_counts_from_last_pipeline <
+                static_cast<uint32_t>(last_bound_desc_set_pipe_layout_state->set_layouts.size());
 
-            if (any_disturbed_bindings) {
+            if (any_disturbed_desc_sets_bindings) {
                 const uint32_t disturbed_bindings_count = static_cast<uint32_t>(
-                    last_bound_desc_set_pipe_layout_state->set_layouts.size() - bindings_counts_from_last_pipeline);
-                const uint32_t first_disturbed_set = bindings_counts_from_last_pipeline;
+                    last_bound_desc_set_pipe_layout_state->set_layouts.size() - desc_set_bindings_counts_from_last_pipeline);
+                const uint32_t first_disturbed_set = desc_set_bindings_counts_from_last_pipeline;
 
                 for (uint32_t set_i = 0; set_i < disturbed_bindings_count; ++set_i) {
                     const uint32_t last_bound_set_i = set_i + first_disturbed_set;
-                    VkDescriptorSet desc_set = last_bound.per_set[last_bound_set_i].bound_descriptor_set->VkHandle();
+                    const auto &last_bound_set_state = last_bound.per_set[last_bound_set_i].bound_descriptor_set;
+                    // last_bound.per_set is a LUT, and descriptor sets before the last one could be unbound.
```

Comment:
> last_bound.per_set is a LUT

I don't think that is accurate, it is an vector of each of the descriptor slots, `set = 4` in your GLSL is `per_set[4]`, not indirect lookup

```suggestion
                    // There can be gaps in the middle of your valid descriptor sets
```

Reply:
I could argue a pointer is a LUT, everything is going to "look up" ... Like I guess I would be willing to call everything a LUT then... Like every state object map is a LUT, there is no end if we start calling thing a LUT

Reply:
No, we do have genuine arrays/list where the index of an element has no real meanings,  and we have maps/LUTs, where the index does have a meaning